packages/api/src/controllers/resources/v1/mod.rs

```diff
@@ -0,0 +1,125 @@
+use by_axum::{
+    auth::Authorization,
+    axum::{
+        extract::{Path, Query, State},
+        routing::get,
+        Extension, Json, Router,
+    },
+};
+
+use models::{
+    // ResourceDeleteRequest,
+    QueryResponse,
+    Resource,
+    ResourceAction,
+    ResourceByIdAction,
+    ResourceCreateRequest,
+    ResourceQuery,
+    ResourceReadAction,
+    ResourceRepository,
+    ResourceSummary,
+    ResourceUpdateRequest,
+};
+
+#[derive(Clone, Debug)]
+pub struct ResourceConterollerV1 {
+    repo: ResourceRepository,
+}
+
+impl ResourceConterollerV1 {
+    pub fn route(pool: sqlx::Pool<sqlx::Postgres>) -> models::Result<Router> {
+        let repo = Resource::get_repository(pool.clone());
+        let ctrl = Self { repo };
+
+        Ok(Router::new()
+            .route(
+                "/:id",
+                get(Self::get_resource).post(Self::act_resource_by_id),
+            )
+            .route("/", get(Self::list_resources).post(Self::act_resource))
+            .with_state(ctrl))
+    }
+    async fn get_resource(
+        State(ctrl): State<ResourceConterollerV1>,
+        Extension(_auth): Extension<Option<Authorization>>,
+        Path(id): Path<String>,
+    ) -> models::Result<Json<Resource>> {
+        let resource = ctrl
+            .repo
+            .find_one(&ResourceReadAction::new().find_by_id(id))
+            .await?;
+        Ok(Json(resource))
+    }
+
+    async fn list_resources(
+        State(ctrl): State<ResourceConterollerV1>,
+        Extension(_auth): Extension<Option<Authorization>>,
+        Query(params): Query<ResourceQuery>,
+    ) -> models::Result<Json<QueryResponse<ResourceSummary>>> {
+        Ok(Json(ctrl.repo.find(&params).await?))
+    }
+
+    async fn act_resource(
+        State(ctrl): State<ResourceConterollerV1>,
+        Extension(_auth): Extension<Option<Authorization>>,
+        Json(body): Json<ResourceAction>,
+    ) -> models::Result<Json<Resource>> {
+        match body {
+            ResourceAction::Create(req) => {
+                let res = Self::create(ctrl.repo.clone(), req).await?;
+                Ok(Json(res))
+            }
```

Comment:
```suggestion
            ResourceAction::Create(req) => ctrl.create(req).await,
```
이런식으로 match arm 을 구성하는게 좋을듯 합니다.